### DISCUSSION TITLE
How do I catch a rejection in Mint?

### DISCUSSION THREAD

**Discussion by nilslindemann** [mint-lang/mint#609] [Answered] [answered]
```mint
let choice =
    await Window.confirm("is this ok?")

// choice is now a Result(String, Void)

// this does not work. The console gives `Uncaught (in promise)` when I press the cancel button
// and `text` will not be `"chancelled"`.
let text =
    case choice {
        Result::Ok => "ok button pressed"
        => "chancelled"
    }
```

How do I pattern match this, so that I get the text "chancelled" when the user presses <kbd>cancel</kbd> and how do I prevent the  `Uncaught (in promise)` error (the uncatched [rejection](https://github.com/mint-lang/mint/blob/24b1e673e74b2300bd9910a3ec076930ba031fb3/core/source/Window.mint#L83))?

**Comment by nilslindemann** [comment, marked as answer]
Hmm, I am starting to believe this is an error, `reject` should be `resolve`. It is the only reject in the code base and [Built-in Types](https://mint-lang.com/guide/reference/built-in-types) says: "In Mint promises have only one parameter result".

**Reply by nilslindemann** [reply]
Made a [PR](https://github.com/mint-lang/mint/pull/610).

**Reply by gdotdesign** [reply]
Yes, you are correct it should have been `resolve` :+1:
